File: 932-monotonic-array/monotonic-array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        
        n = len(nums)
        great = False
        less = False
        for i in range(n-1):
            if (nums[i]<nums[i+1]):
                less = True
            if (nums[i]>nums[i+1]):
                great = True
            if great and less:
                return False
        return True

File: 932-monotonic-array/monotonic-array.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        boolean inc = false;
        boolean dec = false;
        int n = nums.length;
        for(int i=0;i<n-1;i++){

            if(nums[i]>nums[i+1]) inc = true;
            if(nums[i]<nums[i+1]) dec = true;
            if(inc && dec) return false;

        }
        return true;
    }
}